### DISCUSSION TITLE
Best way to handle timestamp with timezone

### DISCUSSION THREAD

**Discussion by pablosat** [prisma/prisma#5445] [Q&A] [answered]
**Context**
Our software needs to show event dates with the timezone of where that event took place. Just like airlines show flights' departure and arrival times with the timezone where the plane takes off and arrives, respectively.

**Constraints**
We are receiving dates with timezone from an API (eg: `2020-12-27T12:00+0800`), but Prisma cannot save dates with timezone, it saves them in UTC.

**Idea**
Save the timezone in an additional column to the related date, to later parallel query the date and its timezone and "manually" do the conversion.

**Question**
Is there a better way to handle this? For example, make a new column `timestamp with timezone` via sql, and somehow query it via Prisma.

pd: we use postgreSQL

**Comment by ryands17** [comment, marked as answer]
@pablosat 👋 

I think with the `nativeTypes` preview feature, it's possible to use `timestampz` in your schema as follows:

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["groupBy", "nativeTypes"]
}

datasource db {
  provider = "postgresql"
  url      = env("DB_URL")
}

model User {
  id   String   @id @default(cuid())
  name String
  date DateTime @db.Timestamptz
}
```

Also note that `timestamptz` saves the date in UTC format as well.

**Reply by pablosat** [reply]
thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
